docs/api-docs.asciidoc

```diff
@@ -0,0 +1,828 @@
+type is ExternalName.
+More info: https://kubernetes.io/docs/concepts/services-networking/service/#virtual-ips-and-service-proxies
+*`type`* _link:https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.13/#servicetype-v1-core[$$Kubernetes core/v1.ServiceType$$]_::
+_(Optional)_
+type determines how the Service is exposed. Defaults to ClusterIP. Valid
+options are ExternalName, ClusterIP, NodePort, and LoadBalancer.
+"ExternalName" maps to the specified externalName.
+"ClusterIP" allocates a cluster-internal IP address for load-balancing to
+endpoints. Endpoints are determined by the selector or if that is not
+specified, by manual construction of an Endpoints object. If clusterIP is
+"None", no virtual IP is allocated and the endpoints are published as a
+set of endpoints rather than a stable IP.
+"NodePort" builds on ClusterIP and allocates a port on every node which
+routes to the clusterIP.
+"LoadBalancer" builds on NodePort and creates an
+external load-balancer (if supported in the current cloud) which routes
+to the clusterIP.
+More info: https://kubernetes.io/docs/concepts/services-networking/service/#publishing-services-service-types
+*`externalIPs`*  _[]string_::
+_(Optional)_
+externalIPs is a list of IP addresses for which nodes in the cluster
+will also accept traffic for this service.  These IPs are not managed by
+Kubernetes.  The user is responsible for ensuring that traffic arrives
+at a node with this IP.  A common example is external load-balancers
+that are not part of the Kubernetes system.
+*`sessionAffinity`* _link:https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.13/#serviceaffinity-v1-core[$$Kubernetes core/v1.ServiceAffinity$$]_::
+_(Optional)_
+Supports "ClientIP" and "None". Used to maintain session affinity.
+Enable client IP based session affinity.
+Must be ClientIP or None.
+Defaults to None.
+More info: https://kubernetes.io/docs/concepts/services-networking/service/#virtual-ips-and-service-proxies
+*`loadBalancerIP`*  _string_::
+_(Optional)_
+Only applies to Service Type: LoadBalancer
+LoadBalancer will get created with the IP specified in this field.
+This feature depends on whether the underlying cloud-provider supports specifying
+the loadBalancerIP when a load balancer is created.
+This field will be ignored if the cloud-provider does not support the feature.
+*`loadBalancerSourceRanges`*  _[]string_::
+_(Optional)_
+If specified and supported by the platform, this will restrict traffic through the cloud-provider
+load-balancer will be restricted to the specified client IPs. This field will be ignored if the
+cloud-provider does not support the feature."
+More info: https://kubernetes.io/docs/tasks/access-application-cluster/configure-cloud-provider-firewall/
+*`externalName`*  _string_::
+_(Optional)_
+externalName is the external reference that kubedns or equivalent will
+return as a CNAME record for this service. No proxying will be involved.
```

Comment:
Avoid future tense.

externalName is the external reference that kubedns or equivalent returns as a CNAME record for this service. No proxying is involved.

Reply:
We can't change this because it comes from source code of the Kubernetes project.